static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.test.tsx

```diff
@@ -75,4 +76,100 @@ describe("UserInfoFormTests", () => {
       "AB12 3CD",
     );
   });
+
+  it("Channel user should be able to edit only card number", () => {
+    global.window = Object.create(window);
+    Object.defineProperty(window, "accountId", { value: "ABCDEF" });
+
+    const initialValues = {
+      name: "Min",
+      marketplace: UserSubscriptionMarketplace.CanonicalProChannel,
+      buyingFor: "organisation",
+      organisationName: "Canonical",
+      address: "ABC Street",
+      city: "DEF City",
+      postalCode: "AB12 3CD",
+      defaultPaymentMethod: {
+        brand: "visa",
+        country: "US",
+        expMonth: 4,
+        expYear: 2024,
+        id: "pm_ABCDEF",
+        last4: "4242",
+      },
+    };
+
+    render(
+      <QueryClientProvider client={queryClient}>
+        <Formik initialValues={initialValues} onSubmit={jest.fn()}>
+          <Elements stripe={stripePromise}>
+            <UserInfoForm setError={jest.fn()} />
+          </Elements>
+        </Formik>
+      </QueryClientProvider>,
+    );
+
+    fireEvent.click(screen.getByRole("button", { name: "Edit" }));
+    fireEvent.change(screen.getByTestId("field-card-number"), {
```

Comment:
nit:

It is recommended to use `userEvent` as it is built on top of `fireEvent`, but it provides several methods that resemble the user interactions more closely. For example, `fireEvent.change` will simply trigger a single change event on the input. However `userEvent.type` will trigger keyDown, keyPress, and keyUp events for each character as well. It's much closer to the user's actual interactions.